src/commands/cmd_tdigest.cc

```diff
@@ -0,0 +1,149 @@
+      auto status = parser.TakeInt<int32_t>();
+      if (!status) {
+        return {Status::RedisParseErr, kErrParseCompression};
+      }
+      auto compression = *status;
+      if (compression <= 0) {
+        return {Status::RedisParseErr, kErrCompressionMustBePositive};
+      }
+      if (compression < 1 || compression > static_cast<int32_t>(kTDigestMaxCompression)) {
+        return {Status::RedisParseErr, kErrCompressionOutOfRange};
+      }
+      options_.compression = static_cast<uint32_t>(compression);
+    }
+
+    return Status::OK();
+  }
+
+  Status Execute(engine::Context &ctx, Server *srv, Connection *conn, std::string *output) override {
+    TDigest tdigest(srv->storage, conn->GetNamespace());
+    bool exists = false;
+    auto s = tdigest.Create(ctx, key_name_, options_, &exists);
+    if (!s.ok()) {
+      if (exists) {
+        return {Status::RedisExecErr, kErrKeyAlreadyExists};
+      }
+      return {Status::RedisExecErr, s.ToString()};
+    }
+    *output = redis::RESP_OK;
+    return Status::OK();
+  }
+
+ private:
+  std::string key_name_;
+  TDigestCreateOptions options_;
+};
+
+class CommandTDigestInfo : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    if (args.size() != 2) {
+      return {Status::RedisParseErr, kErrWrongNumOfInfoArguments};
+    }
+    key_name_ = args[1];
+    return Status::OK();
+  }
+
+  Status Execute(engine::Context &ctx, Server *srv, Connection *conn, std::string *output) override {
+    TDigest tdigest(srv->storage, conn->GetNamespace());
+    TDigestMetadata metadata;
+    auto s = tdigest.GetMetaData(ctx, key_name_, &metadata);
+    if (!s.ok()) {
+      if (s.IsNotFound()) {
+        return {Status::RedisExecErr, kErrKeyNotFound};
+      }
+      return {Status::RedisExecErr, s.ToString()};
+    }
+
+    output->append(conn->HeaderOfMap(9));
+    output->append(redis::BulkString(kInfoCompression));
+    output->append(redis::Integer(metadata.compression));
+    output->append(redis::BulkString(kInfoCapacity));
+    output->append(redis::Integer(metadata.capacity));
+    output->append(redis::BulkString(kInfoMergedNodes));
+    output->append(redis::Integer(metadata.merged_nodes));
+    output->append(redis::BulkString(kInfoUnmergedNodes));
+    output->append(redis::Integer(metadata.unmerged_nodes));
+    output->append(redis::BulkString(kInfoMergedWeight));
+    output->append(redis::Integer(metadata.merged_weight));
+    output->append(redis::BulkString(kInfoUnmergedWeight));
+    output->append(redis::Integer(metadata.total_weight - metadata.merged_weight));
+    output->append(redis::BulkString(kInfoObservations));
+    output->append(redis::Integer(metadata.total_observations));
+    output->append(redis::BulkString(kInfoTotalCompressions));
+    output->append(redis::Integer(metadata.merge_times));
+    output->append(redis::BulkString(kInfoMemoryUsage));
+    output->append(redis::Integer(sizeof(TDigest)));
```

Comment:
It seems this is not so useful, and also, not so accurate. Maybe we can just remove this.

Reply:
Hi @PragmaTwice ,

Thanks for your suggestion! 😊

I have removed this property and left a comment line in code and test case.

Best Regards,
Edward